### PR TITLE
feat(dapps) add DAppsService component and ConnectDAppModal

### LINKED ISSUE #14607: After confirming the URL, the user should see a modal that would show the dApp metadata (name, icon, url) and the decline/connect buttons
- [x] Refactor the DApps wallet connect code to prepare QML code for the upcoming work
  - Move the user workflow responsibility outside the dapp button
  - Create a  dapp workflow component to host the user workflow setup
  - Add a JS service to handle the logic between SDK and presentation layer
- [x] Implement the the dApp Modal for dApp connection display
  - Filter only dapp supported chains
- [x] Pair using the SDK

### PR DESCRIPTION
**Pull request by stefandunca**
### Closes #14607

Implement the initial Pairing user workflow and disconnect option forthe first session.

Also

- rename pairing modal accordingly (`PairWCModal.qml`) to make room for the proper
`ConnectDAppModal.qml`
- basic tests for service helpers
- update storybook to reflect the new user workflows

`HEAD~1`: feat(wallet) customize NetworkFilter for dapps display
`HEAD~2`: chore(dapps) move dApps UX into its own DAppsWorkflow component
`HEAD~3`: chore(dapps) move wallet connect service code outside popups
`HEAD~4`: chore(storybook) load WalletConnect SDK in storybook by QRC for now

### Demo using test dApp at https://react-app.walletconnect.com/

https://github.com/status-im/status-desktop/assets/47554641/730c694a-b69f-414d-82f9-de06fb53a26a

### CODE REVIEW

File: storybook/pages/DAppsWorkflowPage.qml [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+        ColumnLayout {
+            id: optionsSpace
+
+            RowLayout {
+                Text { text: "projectId" }
+                Text {
+                    id: projectIdText
+                    readonly property string projectId: SystemUtils.getEnvVar("WALLET_CONNECT_PROJECT_ID")
+                    text: projectId.substring(0, 3) + "..." + projectId.substring(projectId.length - 3)
```
Comment (saledjenic) [inline storybook/pages/DAppsWorkflowPage.qml:68]: Maybe this instead `text: Utils.elideText(projectId, 3)`?

**Review by saledjenic — APPROVED** [review]

**Review by alexjba — COMMENTED** [review]
Very nice work! 👍 

1. The only blocker for me is that I can't test it and IDK if it's some setup/config I'm missing or it's a bug somewhere.

https://github.com/status-im/status-desktop/assets/47811206/b09ab98a-cdf3-45e3-90be-5ca1d33db301

```
Debug: WC WalletConnectSDK.wcCall.pair; pairLink: wc:465a7cf1a1dc8cf745c20a25e96203a49a8d0a3f32d53be41b96639a4f969000@2?expiryTimestamp=1715890121&relay-protocol=irn&symKey=5936d61cf931e0d831a749d96f7ce3ada9bbef132fa9f2009946b456db20eb6d (qrc:/app/AppLayouts/Wallet/services/dapps/WalletConnectSDK.qml:224, pair)
Debug: WC WalletConnectSDK.onPairResponse; error: Cannot read property 'pair' of null (qrc:/app/AppLayouts/Wallet/services/dapps/WalletConnectSDK.qml:450, onPairResponse)
```


2. These colours are ok in the app? I can't reach this state

Screenshot 2024-05-16 at 23 07 37

**Comment by stefandunca** [comment]
Thanks for the feedback.

> 1. The only blocker for me is that I can't test it and IDK if it's some setup/config I'm missing or it's a bug somewhere.
> 
>  Screen.Recording.2024-05-16.at.23.03.39.mov 
> ```
> Debug: WC WalletConnectSDK.wcCall.pair; pairLink: wc:465a7cf1a1dc8cf745c20a25e96203a49a8d0a3f32d53be41b96639a4f969000@2?expiryTimestamp=1715890121&relay-protocol=irn&symKey=5936d61cf931e0d831a749d96f7ce3ada9bbef132fa9f2009946b456db20eb6d (qrc:/app/AppLayouts/Wallet/services/dapps/WalletConnectSDK.qml:224, pair)
> Debug: WC WalletConnectSDK.onPairResponse; error: Cannot read property 'pair' of null (qrc:/app/AppLayouts/Wallet/services/dapps/WalletConnectSDK.qml:450, onPairResponse)
> ```

The only thing I can think of is the limitation we have that only one instance of status-desktop is working loading the SDK. However, in this case you should see an earlier webengine error that storage can't be accessed. I postponed to look more into this. Please try again, closing the other instances if you have open :(
 
> 2. These colours are ok in the app? I can't reach this state
> 

I reused matching colors from theme, probably they are not working for both states. I will ask design for alternatives.

File: ui/app/mainui/AppMain.qml [hunk 2]
```diff
@@ -2040,4 +2028,27 @@ Item {
...
+    Loader {
+        id: walletConnectServiceLoader
+
+        active: Global.featureFlags.dappsEnabled
+
+        WalletConnectService {
+            id: walletConnectService
+
+            wcSDK: WalletConnectSDK {
+                active: Global.featureFlags.dappsEnabled && WalletStore.RootStore.walletSectionInst.walletReady
```
[the remark this answers is not in the document]
Comment (stefandunca) [reply ui/app/mainui/AppMain.qml:2041]: Indeed this is a workaround to `WalletConnectSDK` that is loaded and starts the `WebEngine` anyway which in turn instantiates the SDK which is not what we want.

[the remark this answers is not in the document]
Comment (stefandunca) [reply ui/app/mainui/AppMain.qml:2041]: Removed this and adding explicit `sourceComponent` fixed the side effect. Thanks for the heads up.

**Review by alexjba — APPROVED** [review]
Nice work! Sorry for the delay

**Pull request merged**